core/src/test/java/org/elasticsearch/cluster/metadata/WildcardExpressionResolverTests.java

```diff
@@ -128,5 +128,17 @@ public void testAll() {
     private IndexMetaData.Builder indexBuilder(String index) {
         return IndexMetaData.builder(index).settings(settings(Version.CURRENT).put(IndexMetaData.SETTING_NUMBER_OF_SHARDS, 1).put(IndexMetaData.SETTING_NUMBER_OF_REPLICAS, 0));
     }
+    
+    public void testForDeprecatedWildCard() {
+      MetaData.Builder mdBuilder = MetaData.builder()
+          .put(indexBuilder("test").state(IndexMetaData.State.OPEN))
+          .put(indexBuilder("test1").state(IndexMetaData.State.OPEN));
+      ClusterState state = ClusterState.builder(new ClusterName("_name")).metaData(mdBuilder).build();
+      IndexNameExpressionResolver.WildcardExpressionResolver resolver = new IndexNameExpressionResolver.WildcardExpressionResolver();
+
+      IndexNameExpressionResolver.Context context = new IndexNameExpressionResolver.Context(state, IndicesOptions.fromOptions(true, true, true, true));
+      assertThat(newHashSet(resolver.resolve(context, Arrays.asList("+test*"))), equalTo(newHashSet("test", "test1")));
```

Comment:
nit: can you also add here the same few tests that you had above, which combined the use of '+' with other names in the same expression